src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -2079,13 +2075,13 @@ private void InvokeCallbacks(
 			for (var callbackIndex = 0; callbackIndex < currentCallbacks.Length; callbackIndex++)
 			{
 				var callback = currentCallbacks[callbackIndex];
-				eventArgs ??= new DependencyPropertyChangedEventArgs(property, previousValue, newValue
-#if __APPLE_UIKIT__ || IS_UNIT_TESTS
-					, previousPrecedence, newPrecedence, bypassesPropagation
-#endif
-				);
 				callback.Invoke(instanceRef, property, eventArgs);
 			}
+
+			eventArgs.OldValueInternal = null;
+			eventArgs.NewValueInternal = null;
```

Comment:
Do we need to clear them here? They will be overwritten next time they're borrowed.

Reply:
Let's explain why it's there (avoid leaks)